packages/teleport/src/Console/consoleContext.tsx

```diff
@@ -115,34 +120,37 @@ export default class ConsoleContext {
   }
 
   refreshParties() {
-    // Finds unique clusterIds from all active ssh sessions
-    // and creates a separate API call per each.
-    // After receiving the data, it updates the stores only once.
-    const clusters = this.storeDocs
-      .getSshDocuments()
-      .filter(doc => doc.status === 'connected')
-      .map(doc => doc.clusterId);
-
-    const unique = [...new Set(clusters)];
-    const requests = unique.map(clusterId =>
-      // Fetch parties for a given cluster and in case of an error
-      // return an empty object.
-      serviceSession.fetchParticipants({ clusterId }).catch(err => {
-        logger.error('failed to refresh participants', err);
-        const emptyResults: ParticipantList = {};
-        return emptyResults;
-      })
-    );
-
-    return Promise.all(requests).then(results => {
-      let parties: ParticipantList = {};
-      for (let i = 0; i < results.length; i++) {
-        parties = {
-          ...results[i],
-        };
-      }
-
-      this.storeParties.setParties(parties);
+    return tracer.startActiveSpan('refreshParties', span => {
+      // Finds unique clusterIds from all active ssh sessions
+      // and creates a separate API call per each.
+      // After receiving the data, it updates the stores only once.
+      const clusters = this.storeDocs
+        .getSshDocuments()
+        .filter(doc => doc.status === 'connected')
+        .map(doc => doc.clusterId);
+
+      const unique = [...new Set(clusters)];
+      const requests = unique.map(clusterId =>
+        // Fetch parties for a given cluster and in case of an error
+        // return an empty object.
+        serviceSession.fetchParticipants({ clusterId }).catch(err => {
+          logger.error('failed to refresh participants', err);
+          const emptyResults: ParticipantList = {};
+          return emptyResults;
+        })
+      );
+
+      return Promise.all(requests).then(results => {
+        let parties: ParticipantList = {};
+        for (let i = 0; i < results.length; i++) {
+          parties = {
+            ...results[i],
+          };
+        }
+
+        this.storeParties.setParties(parties);
```

Comment:
This logic is equivalent to the following suggestion:
Only the last element of the results is assigned to the parties
```suggestion
        this.storeParties.setParties(results[results.length-1]);
```
Or should this merge results into one object? :
```suggestion
        const parties: ParticipantList = {};
        for (const result in results) {
            Object.assign(parties, result);
        }
        this.storeParties.setParties(parties);
```

Reply:
Thanks @JanKaczmarkiewicz but that was just lift & shifted by adding the telemetry to it, so at this time I don't want to introduce any more changes 😄 but I'll make a note to revisit it.